fasthttpproxy/example_test.go

```diff
@@ -0,0 +1,112 @@
+package fasthttpproxy
+
+import (
+	"log"
+	"time"
+
+	"github.com/valyala/fasthttp"
+	"golang.org/x/net/http/httpproxy"
+)
+
+func ExampleDialer_GetDialFunc() {
+	p := Dialer{
+		TCPDialer: fasthttp.TCPDialer{
+			Concurrency: 1000,
+		},
+		Timeout:        time.Second * 30,
+		ConnectTimeout: time.Second * 20,
+		Config: httpproxy.Config{
+			HTTPProxy:  "http://127.0.0.1:8005",
+			HTTPSProxy: "http://127.0.0.1:8005",
+		},
+		DialDualStack: true,
+	}
+	// proxy information comes from environment
+	dialer, err := p.GetDialFunc(false)
+	if err != nil {
+		log.Fatal(err)
+	}
+	client := &fasthttp.Client{
+		ReadTimeout:         time.Second * 5,
+		WriteTimeout:        time.Second * 5,
+		MaxIdleConnDuration: time.Second * 5,
+		Dial:                dialer,
+	}
+	statusCode, resp, err := client.Get(nil, "https://www.google.com")
+	if err != nil {
+		log.Fatal(err)
+	}
+	if statusCode != 200 {
+		log.Fatal(err)
+	}
+	_ = resp
+}
+
+func ExampleDialer_GetDialFunc_second() {
+	p := Dialer{
+		TCPDialer: fasthttp.TCPDialer{
+			Concurrency: 1000,
+		},
+		Config: httpproxy.Config{
+			HTTPProxy:  "http://127.0.0.1:8005",
+			HTTPSProxy: "http://127.0.0.1:8005",
+			NoProxy:    "example.com",
+		},
+		Timeout:        time.Second * 30,
+		ConnectTimeout: time.Second * 20,
+		DialDualStack:  true,
+	}
+	dialer, err := p.GetDialFunc(false)
+	if err != nil {
+		log.Fatal(err)
+	}
+	client := &fasthttp.Client{
+		ReadTimeout:         time.Second * 5,
+		WriteTimeout:        time.Second * 5,
+		MaxIdleConnDuration: time.Second * 5,
+		Dial:                dialer,
+	}
+	// this request doesn't use proxy.
+	statusCode, resp, err := client.Get(nil, "https://example.com")
+	if err != nil {
+		log.Fatal(err)
+	}
+	if statusCode != 200 {
+		log.Fatal(err)
+	}
+	_ = resp
+}
+
+func ExampleDialer_GetDialFunc_three() {
```

Comment:
What is the difference between this and the second example? Besides `NoProxy: "example.com",`

Reply:
My thinking was somewhat flawed; I should have combined the three examples together. Thank you.